### ISSUE TITLE
Add support for rtd1296

### ISSUE THREAD

**Issue by bidorffOL** [SynoCommunity/spksrc#2886]
Because of the new rtd1296 architecture supported by the 418J, I cannot add the SynoCommunity repository to my NAS. From what I have read in the [FAQ](https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions#i-have-a-new-nas-model-what-has-to-be-done-to-provide-packages-for-it), one of the prerequisite for such support is the availability of the toolchain for this architecture (which has been added yesterday by Synology [here](https://sourceforge.net/projects/dsgpl/files/DSM%206.1%20Tool%20Chains/Intel%20x86%20Linux%204.4.15%20%28Rtd1296%29/)).

Is there any hope to see this new architecture supported in a near future ?

**Comment by davebooi** [comment]
I second this and posted a 30 bucks bounty on [bountysource](https://www.bountysource.com/issues/48574365-add-support-for-rtd1296)

**Comment by Diaoul** [comment]
Off topic: This is really annoying that Bountysource doesn't update the issue... It suddenly stopped working, I'll get in touch with them.

**Comment by ymartin59** [comment]
Adding toolchain in spksrc is a five minute job and it should be enough for anyone to build its own packages... but job to get any DSM 6.1 package available through Package Center is still long: https://github.com/SynoCommunity/spksrc/wiki/Frequently-Asked-Questions#which-dsm-versions-do-you-support

**Comment by bidorffOL** [comment]
I actually tried to do it myself on my own [fork](https://github.com/bidorffOL/spksrc/) and I managed to get NZBGet and GIT to build and run, still no luck on Python, but I'm not sure if the toolchain is responsible for the broken build.

I'll try to provide a Pull Request for my changes (I will need to filter those for the toolchains and those for the packages I updated and / or modified).

**Comment by albino71** [comment]
@bidorffOL Got me a DS418J..great device, but the nzbget package is not working and cannot be installed...

Any chance you could slip me your installable package?

**Comment by bidorffOL** [comment]
You can try [this version](https://github.com/bidorffOL/syno-packages/raw/master/nzbget_rtd1296-6.1_19.1-23.spk) (tell me if it works, I'm not sure that this is the file I actually used).

**Comment by bidorffOL** [comment]
That's what I was afraid of...

I'll try to reinstall my development environment this weekend and provide you with a proper build, otherwise you may try yourself to setup your own environment using my own [fork](https://github.com/bidorffOL/spksrc) and the `dsm6` branch (this should work).

**Comment by albino71** [comment]
@bidorffOL No problem...
I'm a bit of a noob when it comes to this, but maybe I will give it a try...
Thanks for your time and effort anyway!

**Comment by bidorffOL** [comment]
I updated the file on my repository with a fresh build, can you try again ?

**Comment by Keeviiin** [comment]
Hello,
I got my new DS418 yesterday and as you know, it uses the rtd1296-Architecture as well. I am totally new to the compiling-stuff an I have no spksrc running. I dont have enough knowledge to get a spksrc running so could you please help an provide me one package for rtd1296: ffmpeg

Thanks a lot in advance :)

**Comment by ymartin59** [comment]
I tried to compile current ffmpeg version to apollolake toolchain but there is ARMv8 / AARCH64 specific issue with assembly compilation...
x264 assembly source code is specific to Apple compiler for iOS/macOS, a pity team ignores GNU linaro.
So `ffmpeg` will not be there soon.

**Comment by fbernardo** [comment]
@ymartin59 Could you also give python a try? I’ve been trying to compile it but always hit a problem related with cffi. But I might be doing something wrong since I got the same error when compiling for x64.

**Comment by Keeviiin** [comment]
Sorry that I might ask obvious stuff:
So you say that it is not just a single compiling task to get a working ffmpeg package? It is such a lot of work that is not sure if it even works in half a year or so..?

Did you try to compile it for rtd1296?

Thank you very much.

**Comment by ymartin59** [comment]
Yes compiling with rtd1296 toolchain does not work at the moment. Either assembly code has to be ported to linary to get hardware optimization, either I have to fix linking troubles (with patches) that appeared with that new toolchain... I will probably try this second option first but it is a pity x264 hardware acceleration is lacking on ARM whereas assembly code is already available (but not for gcc compiler)

**Comment by Keeviiin** [comment]
That is very sad to hear, so I will have to keep on waiting :)
Thank you very much for your efforts!

**Comment by WodKa78** [comment]
Also successfully installed nzbget thanks to [bidorffol](https://github.com/bidorffol) but some other packages i also would love to install and run (Sonarr and Radarr) but they are not available for the ds418j. Is it possible somebody makes them available or is this not possible due the hardware limitations?

**Comment by banshee1971** [comment]
Im waiting too! But Meanwhile, im running Sonar and Sabnzb on a dedicated PC until new package out!

**Comment by WodKa78** [comment]
I tried to compile them myself, using [bidorffol](https://github.com/bidorffol) fork and dsm6 branch but constantly ending with errors. guess this is due to my enviorment since none of the packages i tried to compile were succesfully created. I will have to study some more i guess....

**Comment by WodKa78** [comment]
some working packages i posted here https://github.com/SynoCommunity/spksrc/issues/2971

**Comment by ymartin59** [comment]
As owner of DSM rtd1296 architecture, may you please download and test this "beta" `ffmpeg` built: http://dl.free.fr/oIHqgBdrA
If possible, please run from command line `ffmpeg -i` on various media files. 
Warning: do not expect Video Station to use it, as recent version is stuck to Synology embedded version.
Please report success or any failure with command output

**Comment by piejanssens** [comment]
@ymartin59 Any particular reason nzbget is not in the repo for rtd1296 and neither in your test package list found here? https://github.com/SynoCommunity/spksrc/issues/3138

**Comment by WodKa78** [comment]
I also noticed its not visible in the package center anymore, no idea why but you can download it [here](https://synocommunity.com/package/nzbget) and install manually . Update from within Nzbget.

**Comment by ymartin59** [comment]
Hello. Package Center probably has trouble with cache - known issue at Synology. Try to force "refresh" and see if packages appear.
Avoid to comment on close issues... I have far enough to only follow opened ones.
